client/src/routes/game/+page.svelte

```diff
@@ -0,0 +1,68 @@
+<script lang="ts">
+  import LobbyComponent from "$components/Lobby.svelte";
+  import { onSnapshot, doc, getDoc } from "firebase/firestore";
+  import { onMount } from "svelte";
+  import { lobbyCollection } from "$lib/firebase/firestore-collections";
+  import type { Lobby } from "$lib/firebase/firestore-types/lobby";
+  import { page } from "$app/stores";
+  import { authStore as user } from "$stores/auth";
+  import { goto } from "$app/navigation";
+  import type { Unsubscribe } from "firebase/auth";
+
+  let lobbyData: Lobby;
+  let code: string;
+  // in case we want to unsub from lobby
+  let unsubscribeLobby: Unsubscribe;
+
+  onMount(async () => {
+    // gets code from url search
+    // the svelte magic with searchparams wasnt working
+    code = $page.url.search.split("code=")[1];
```

Comment:
this will not work with a url that looks like `/game?code=abcxyz&src=share-button`

why not use something like this?
```ts
$page.url.searchParams.get("code")
```